request/do.go

```diff
@@ -0,0 +1,101 @@
+package request
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+)
+
+func doRequest(url string, timeout time.Duration) Record {
+	client := http.Client{
+		// Timeout includes connection time, any redirects, and reading the response body.
+		// We may want exclude reading the response body in our benchmark tool.
+		Timeout: timeout,
+	}
+
+	t0 := time.Now()
+
+	resp, err := client.Get(url)
+	if err != nil {
+		return Record{error: fmt.Sprint(err)}
+	}
+
+	return newRecord(resp, time.Since(t0))
```

Comment:
We shouldn't stop the timer at the end of the function, but rather as soon as the request is over so we don't include our own error handling in it (and potentially more if this function gets updated) :
```suggestion
	t0 := time.Now()
	
	resp, err := client.Get(url)
	elasped := time.Since(t0)
	if err != nil {
		return Record{error: fmt.Sprint(err)}
	}

	return newRecord(resp, elapsed)
```